[lynnetteeee] iP

lynnetteeee opened the pull request:
# Floofy
>  Floofy is here to ensure your tasks are REMEMBERED, especially when your brain FLOOFS to wonderland!

## Why should you use Floofy? It's 
*  friendly :smiling_face_with_three_hearts:	
* cute :point_right:	:point_left:
* ~~SPEEDY~~ SUPER SPEEDY to use

## First things first! 
1. Download Java 11 [here](https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html)
2. Collate your tasks
3. Start using Floofy!

## Floofy's Features:
- [x] Basic: Add/Delete/Mark/Unmark tasks
- [x] Enhanced: Find/List all tasks 
- [x] Manage Tasks Types: ToDo, Deadline, Event
- [ ] Set Reminders (coming soon)

### For Java programmers, here's a sneak peek of how YOUR `Floofy` object is actually created!
> Yes, you see a __unique__ `Ui` that only our dearest `Floofy` has!
```java
public Floofy(String filePath) {
        ui = new Ui();
        storage = new Storage(filePath);
        parser = new Parser();
        try {
            tasks = new TaskList();
            storage.loadTasks(tasks);
        } catch (FloofyException e) {
            ui.showLoadingError();
            tasks = new TaskList();
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

lynnetteeee committed:
finish level 0

lynnetteeee committed:
finish level-1

lynnetteeee committed:
finish level 2

lynnetteeee committed:
finish level 3, added new class Task.java

lynnetteeee committed:
finish level 4, added classes ToDos, Deadline, Event

lynnetteeee committed:
added testing

lynnetteeee committed:
basic exceptions for invalid format (not including empty fields yet)

lynnetteeee committed:
added delete feature

lynnetteeee committed:
added another checked exception

lynnetteeee committed:
save feature

lynnetteeee committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  save feature

AidenLYT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,204 @@
+        loadTasks(list);
+        String line = "__________________________________________________";
+        System.out.println(line);
+        System.out.println("Hello! I'm Floofy");
```
Since your ChatBot is named Floofy, it will be better to change the class filename from Duke.java to Floofy.java 😄

AidenLYT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,204 @@
+                list.add(newTodo);
+                newTodo.addTask(list.size());
+                System.out.println(line);
+                saveTasks(list);
```
Do try to practice abstraction. best not to show the detailed implementation, but only show the functionality 😄

AidenLYT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,204 @@
+                list.add(newDeadline);
+                newDeadline.addTask(list.size());
+                System.out.println(line);
+                saveTasks(list);
```
Practice abstraction here too

AidenLYT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,204 @@
+                list.add(newEvent);
+                newEvent.addTask(list.size());
+                System.out.println(line);
+                saveTasks(list);
```
Practice abstraction here as well

AidenLYT commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public String toFileFormat() {
+        return String.format("%s | %d | %s", this.getType(), this.isDone ? 1 : 0, this.description);
```
Do practice SLAP Hard code quality (refer to https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html) 

````
suggestion: 
int taskDone = this.isDone ? 1 : 0;
return String.format("%s | %d | %s", this.getType(), taskDone, this.description);
````

AidenLYT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,204 @@
+            if (file.getParentFile() != null && !file.getParentFile().exists()) {
+                if (!file.getParentFile().mkdirs()) {
+                    System.err.println("Failed to create parent directories.");
+                    return;
```
Can the two 'if'-statement be merged into one?

AidenLYT commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+    protected String from;
+    protected String to;
+
+    public Event(String description, String eventFrom, String eventTo) {
```
It would be good practice to include comments for public classes/method. 
Code Quality - Guideline: Comment minimally, but sufficiently (https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html)

AidenLYT reviewed (commented):
Overall, I wish that you would practice abstraction in your code (hiding implementation, showing functionality). Consider splitting into smaller classes. Nonetheless, good attempt for the ChatBot 👍🏻 Logic is easy to follow.

lynnetteeee committed:
level 8 date done

lynnetteeee committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  level 8 date done

lynnetteeee committed:
extract ui class [A-MoreOOP]

legionlegion commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,224 @@
+        String[] parts = line.split(" \\| ");
+        Task task = null;
+
+        switch (parts[0]) {
```
Please change the indentation for switch and case statements, as there is no indentation for case clauses.

legionlegion commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,224 @@
+                    continue;
+                }
+                System.out.println(line);
+                int preIdxFrom = userInput.indexOf("/from");
```
Perhaps some of the upcoming lines can be separated by one blank line, such as between variable definitions and calling methods.
e.g. 
```java
                LocalDate eventDateTo = LocalDate.parse(eventTo);
                Event newEvent = new Event(eventTask, eventDateFrom, eventDateTo);

                list.add(newEvent);
                newEvent.addTask(list.size());

                System.out.println(line);
```

legionlegion commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+import java.time.LocalDate;
+import java.time.format.TextStyle;
+import java.util.Locale;
+public class Deadline extends Task {
```
Perhaps it would be better formatted if you keep a line between the imports and your class definition?

legionlegion commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,224 @@
+
+        switch (parts[0]) {
+            case "T":
+                task = new ToDos(parts[2]);
```
I like how you added comments to most of your switch cases, perhaps it would be good to add the case of "T" as well.

legionlegion reviewed (commented):
LGTM! Just some nits to fix!

yashma-sonara commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,224 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
```
You could consider adding javadocs and descriptive headers to your public classes and methods.

yashma-sonara commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception {
```
Perhaps you can add subclasses of exceptions (ex. empty todo description, missing time), so you don't have to make a new Exception every time in Duke.java

yashma-sonara commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.format.TextStyle;
+import java.util.Locale;
+public class Event extends Task {
+    protected LocalDate from;
```
Is it necessary your variables to have access modifier: protected as opposed to private. As you have not implemented package, having it as protected doesn't provide additional utility?

yashma-sonara commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,32 @@
+        this.to = eventToDate;
+    }
+
+    public String getDateString(LocalDate date) {
+        return date.getMonth().getDisplayName(TextStyle.SHORT, Locale.ENGLISH) + " " + date.getDayOfMonth() + " " + date.getYear();
+    }
```
I like how you are using TextStyle to handle the date display.

yashma-sonara commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,7 @@
+list
+mark 2
+list
+bye
```
Maybe you can test out the delete function as well?

yashma-sonara reviewed (commented):
For the levels completed, LGTM. Just some nits to fix.

lynnetteeee committed:
fixed TaskList.java, Ui.java, Storage.java for MoreOOP

lynnetteeee committed:
Merge branch 'branch-increments-MoreOOP'

* branch-increments-MoreOOP:
  fixed TaskList.java, Ui.java, Storage.java for MoreOOP
  extract ui class [A-MoreOOP]

lynnetteeee committed:
changed all "Duke" to "Floofy"

lynnetteeee committed:
packaged into floofy, further packaged tasks into floofy.task

lynnetteeee committed:
Merge branch 'branch-increments-Packages'

* branch-increments-Packages:
  packaged into floofy, further packaged tasks into floofy.task
  changed all "Duke" to "Floofy"

lynnetteeee committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

lynnetteeee committed:
updated gradle plugins, able to run main class

lynnetteeee committed:
created 2 test files, changed ToDos to ToDo

lynnetteeee committed:
Merge branch 'branch-increments-JUnit'

* branch-increments-JUnit:
  created 2 test files, changed ToDos to ToDo

lynnetteeee committed:
no changes to code, only renamed archiveBaseName in shadowJar of build.gradle

lynnetteeee committed:
Merge branch 'branch-increments-Jar'

* branch-increments-Jar:
  no changes to code, only renamed archiveBaseName in shadowJar of build.gradle

lynnetteeee committed:
added javadocs for 7/10 classes

lynnetteeee committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  added javadocs for 7/10 classes

lynnetteeee committed:
Change size() method in TaskList to getSize()

lynnetteeee committed:
Add find feature

lynnetteeee committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Change size() method in TaskList to getSize()

lynnetteeee committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find feature

lynnetteeee committed:
Add JavaFx Tutorial Part 1

lynnetteeee committed:
Add JavaFx Tutorial Parts 2 and 3

lynnetteeee committed:
Add JavaFx Tutorial Part 4

lynnetteeee committed:
Remove JavaFx Tutorial Parts 1-3 Code From Floofy.java

lynnetteeee committed:
Change all Ui methods to return String

lynnetteeee committed:
Integrate chatbot features with GUI

lynnetteeee committed:
Clean up code

lynnetteeee committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Clean up code
  Integrate chatbot features with GUI
  Change all Ui methods to return String
  Remove JavaFx Tutorial Parts 1-3 Code From Floofy.java
  Add JavaFx Tutorial Part 4
  Add JavaFx Tutorial Parts 2 and 3
  Add JavaFx Tutorial Part 1

lynnetteeee committed:
Rename data/duke.txt to data/floofy.txt

lynnetteeee committed:
Update Floofy Naming

lynnetteeee committed:
Update build.gradle before testing whether assertions work

The mainClass of the application was initially set to my main class,
which was where the program run from before the GUI was set in.
I have updated it to be the Launcher now that it is the entry point
of this program.

As I will be using gradle to run my program and check if the assert
statements work, I had to add the `run {enableAssertions = true}` to the file.

By ensuring the build.gradle file is updated, it can then allow me to work
on the assert statements thereafter.

lynnetteeee committed:
Update storage path to floofy.txt

lynnetteeee committed:
Add assert statements

Assert statements were added to verify the assumption that a task number
is greater than 0 when the command involves directly typing in a task
number, specifically the "mark" and "unmark" commands.

An assert false statement has also been added to the last default case of the
switch case statements where an invalid command has been used.

With these assert statements, future modifications to e.g. the Parser or in processing
the indices of the commands can be more visible to the developer as AssertionErrors
with the respective messages will be reflected in the console.

lynnetteeee committed:
Shortened Floofy switch-case code

Lines that are repeated in different cases of a switch statement
needs to be removed to allow the code to be more organised, and to
reduce the length of the code that readers need to read

Let's remove the line "storage.saveTasks(tasks)" from multiple cases
in switch-case and place it in the finally block of the code

This is done to also improve on the logical flow of the code, as the
finally clause does the "tidying up" after all operations are done,
which in this case, it would be more intuitive to put the function that
saves the tasks after each command in the finally clause.

lynnetteeee committed:
Floofy class: remove unnecessary variable in runCommand function

This method returns a String, and used a variable to store the String
that is to be returned for each case, before finally returning that
variable at the end of the switch-case statement. This can cause users
to potentially 'trip-up' due to the anomaly in the data flow, since
there is no reason to assign the String to a variable before returning.

Let's remove the variable and return the String directly in each case.

This shortens the code, and improves the readability of the code as
readers no longer need to trace all the way to the bottom of the
switch-case method to find where the return statement is. It also
removes the need for break statements.

lynnetteeee committed:
Parser class: replace default invalid block in parse method

The parse method parses the user input to determine the command. The
else case in the method assigns the parsed input to "invalid",
which is a slight data flow anomaly. This ultimately affects the
organisation of the runCommand() function in Floofy class, which
had to contain a case for a command named "invalid". This potentially
confuses readers.

Lets replace the assigning of the parsed input to "invalid" with
throwing FloofyException, which was created to be thrown when invalid
inputs are given by the user.

This allows us to make use of the behaviour of an exception, and also
removes the need to return an empty String in the runCommand() method in
Floofy, since without an "invalid" parsed input, the default case can be
used for invalid commands to simply throw the exception.

lynnetteeee committed:
Apply SLAP to parse method in Parser.java, shorten too-long method

lynnetteeee committed:
Add javadocs for refactored parser
Fix minor bug that only allows single-digit tasks index to be input

lynnetteeee committed:
Shorten lines in Parser.java with line length > 120 chars

lynnetteeee committed:
Merge branch 'master' into branch-A-CodeQuality

lynnetteeee committed:
Merge pull request #2 from lynnetteeee/branch-A-CodeQuality

Refactor code to improve code quality

lynnetteeee committed:
Merge branch 'master' into branch-A-Assertions

lynnetteeee committed:
Merge pull request #3 from lynnetteeee/branch-A-Assertions

Add assert statements

lynnetteeee committed:
Parser class: Improve code quality

This is to prepare the code for C-Friendlier Syntax. The parser uses
absolute indices to access several components of the user's input,
which would not work if we were to allow for aliases.

Let's remove the use of magic numbers, with integers only used to access
relative index.

Let's also improve the readability of the each command's parser method
by abstracting out details of parsing into smaller helper methods.

These allow the code to be more easily extendable in future when a
greater variety of commands are accepted or when new commands
are created.

lynnetteeee committed:
Add at least 1 friendlier syntax for each command

lynnetteeee committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Add at least 1 friendlier syntax for each command
  Parser class: Improve code quality

lynnetteeee committed:
Update build.gradle shadowJar configurations

lynnetteeee committed:
Update Parser methods to check commands, use .equals to compare command

lynnetteeee committed:
Add Ui.png, set product name in title bar of window

lynnetteeee committed:
Add User Guide

lynnetteeee committed:
Update README

lynnetteeee committed:
Check and update javadocs - Task, ToDo, Deadline, Event

lynnetteeee committed:
Abstracted getDateString method from Event and Deadline to Task class

lynnetteeee committed:
Update import statements in Deadline and Event

lynnetteeee committed:
Refactor Floofy: abstract from runCommand(), adhere to SLAP

lynnetteeee committed:
Refactor Floofy: add javadocs for new methods

lynnetteeee committed:
Add clause to close application with bye command

lynnetteeee committed:
Update Parser: add javadocs

lynnetteeee committed:
Update Storage: add javadocs

lynnetteeee committed:
Update Storage: Error handling of loadTasks

lynnetteeee committed:
Refactor Storage: abstract from parseTask, adhere SLAP + add javadocs

lynnetteeee committed:
Format FloofyException: add white line

lynnetteeee committed:
Update Ui: add javadocs

lynnetteeee committed:
Update Ui: change line field to local variable in wrapWithLines

lynnetteeee committed:
Format Ui: remove white line

lynnetteeee committed:
Update TaskList: add javadocs

lynnetteeee committed:
Update DeadlineTest: add javadocs

lynnetteeee committed:
Update ToDoTest: add javadocs

lynnetteeee committed:
Update Floofy: add exceptions to check for invalid idx

lynnetteeee committed:
Update Floofy: add javadocs for new methods
